Source/Shared/Paths.cs

```diff
@@ -3,112 +3,186 @@
 
 namespace CodeImp.Bloodmasters;
 
-public static class Paths
+internal enum StartupMode
 {
-    private const string DevModeMarkerFileName = ".bloodmasters.dev.marker";
-    private const string DevSolutionRootMarkerFileName = ".bloodmasters.solution-root.marker";
-    private const string TargetFrameworkForExecutables = "net7.0-windows";
-    private const string BuildConfiguration =
-#if DEBUG
-        "Debug";
-#else
-        "Release";
-#endif
+    AutoDetect,
+    Production,
+    Dev
+}
+
+public abstract class Paths
+{
+    public const string GameName = "Bloodmasters";
+    public const string DefaultConfigFileName = "Bloodmasters.cfg";
+
+    protected const string ClientExecutableFileName = "Bloodmasters.exe";
+    protected const string LauncherExecutableFileName = "BMLauncher.exe";
 
     /// <summary>Directory of the entry binary.</summary>
     /// <remarks>In dev mode, it looks like <c>Source/Client/bin/Debug/net7.0-windows</c>.</remarks>
-    private static readonly string AppBaseDir = GetAppBaseDir();
-    private static string GetAppBaseDir()
+    protected static readonly string AppBaseDir = EvaluateAppBaseDir();
+
+    private static string EvaluateAppBaseDir()
     {
         var modulePath = Assembly.GetEntryAssembly()?.Location ?? Process.GetCurrentProcess().MainModule?.FileName;
-        if (modulePath == null) return Environment.CurrentDirectory;
+
+        if (modulePath is null)
+        {
+            return Environment.CurrentDirectory;
+        }
+
         return Path.GetDirectoryName(modulePath) ?? Environment.CurrentDirectory;
     }
 
-    internal static readonly bool IsDevModeBuild =
-        File.Exists(Path.Combine(AppBaseDir, DevModeMarkerFileName));
+    internal static Paths Create(StartupMode startupMode)
+        => startupMode switch
+        {
+            StartupMode.AutoDetect when DevPaths.HasDevModeMarker => new DevPaths(),
+            StartupMode.AutoDetect or StartupMode.Production => new ProductionPaths(),
+            StartupMode.Dev => new DevPaths(),
```

Comment:
_(Minor)_ I'd find this to be more readable:
```suggestion
            StartupMode.AutoDetect => DevPaths.HasDevModeMarker ? new DevPaths() : new ProductionPaths(),
            StartupMode.Production => new ProductionPaths(),
            StartupMode.Dev => new DevPaths(),
```